lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -273,52 +262,16 @@ sub do_asset_caching {
     return undef;
 }
 
-sub engine_workit {
-    my ($job)           = @_;
-    my $worker          = $job->worker;
-    my $client          = $job->client;
-    my $global_settings = $worker->settings->global_settings;
-    my $pooldir         = $worker->pool_directory;
-    my $instance        = $worker->instance_number;
-    my $workerid        = $client->worker_id;
-    my $webui_host      = $client->webui_host;
-    my $job_info        = $job->info;
-
-    log_debug('Preparing Mojo::IOLoop::ReadWriteProcess::Session');
-    session->enable;
-    session->reset;
-    session->enable_subreaper;
-
-    my ($sysname, $hostname, $release, $version, $machine) = POSIX::uname();
-    log_info('+++ setup notes +++', channels => 'autoinst');
-    log_info(sprintf("Running on $hostname:%d ($sysname $release $version $machine)", $instance),
-        channels => 'autoinst');
-
-    log_error("Failed enabling subreaper mode", channels => 'autoinst') unless session->subreaper;
-
-    # XXX: this should come from the worker table. Only included
-    # here for convenience when looking at the pool of
-    # debugging.
-    my $job_settings = $job_info->{settings};
-    for my $i (qw(QEMUPORT VNC OPENQA_HOSTNAME)) {
-        $job_settings->{$i} = $ENV{$i};
-    }
-    if (open(my $fh, '>', 'job.json')) {
-        print $fh Cpanel::JSON::XS->new->pretty(1)->encode($job_info);
-        close $fh;
-    }
-
+sub prepare_environment ($webui_host, $instance, $worker_id, $global_settings, $job, $pooldir) {
     # pass worker instance and worker id to isotovideo
     # both used to create unique MAC and TAP devices if needed
     # workerid is also used by libvirt backend to identify VMs
-    my $openqa_url = $webui_host;
-    my %vars       = (
-        OPENQA_URL      => $openqa_url,
+    my %vars = (
+        OPENQA_URL      => $webui_host,
         WORKER_INSTANCE => $instance,
-        WORKER_ID       => $workerid,
+        WORKER_ID       => $worker_id,
         PRJDIR          => OpenQA::Utils::sharedir(),
-        %$job_settings
-    );
+        %$job->info->{settings});
```

Comment:
this triggers `Can't locate object method "info" via package "21"`. Anyone has an idea how to solve that better?

Reply:
```suggestion
        %{ $job->info->{settings} })
```

Reply:
Right, that gives me `Use of uninitialized value in concatenation (.) or string`

Reply:
there is no string concatenation happening in this part of the code?

Reply:
I was wondering about the same. Generally the suggested code change is supposed to work.

Reply:
ah ok, I see that it's now about a different line. Well, that helps, I can go on :)

Reply:
Well, couldn't understand that part while being in another casual video meeting so for now I just left out that commit :)